src/ActivityRepository.js

```diff
@@ -0,0 +1,61 @@
+const UserActivity = require('../src/UserActivity');
+const activitySamples = require('../data/activitySamples');
+
+class ActivityRepository {
+  constructor() {
+    this.activityInformation = activitySamples;
+  };
+
+  obtainActivityUser(id) {
+    return this.activityInformation.filter(userInfo => {
+      if (userInfo.userID === id) {
+        return userInfo;
+      }
+    })
+  };
+
+  instantiateUserActivity(id) {
+    return new UserActivity(this.obtainActivityUser(id));
+  };
+
+  findAverageFlightsOfStairsClimbedForADate(date) {
+    let specifiedDate = this.findDateForActivity(date);
+    let totalFlightsOfStairsClimbed = specifiedDate.reduce((totalStairs, day) => {
+      return totalStairs += day.flightsOfStairs;
+    }, 0)
+    return Math.floor(totalFlightsOfStairsClimbed / specifiedDate.length)
+  };
+
+  findAverageNumberOfStepsTakenForADate(date) {
+    let specifiedDate = this.findDateForActivity(date);
+    let totalNumberOfStepsTaken = specifiedDate.reduce((totalSteps, day) => {
+      return totalSteps += day.numSteps;
+    }, 0)
+    return Math.floor(totalNumberOfStepsTaken / specifiedDate.length)
+  };
+
+  findAverageMinutesActiveForADate(date) {
+    let specifiedDate = this.findDateForActivity(date);
+    let totalMinutesActive = specifiedDate.reduce((totalMins, day) => {
+      return totalMins += day.minutesActive;
+    }, 0)
+    return Math.floor(totalMinutesActive / specifiedDate.length);
+  };
+
+  findLaziestPersonForADate(date) {
```

Comment:
If I didn't participate in this project, I might imagine this function was supposed to help the user of the function find the laziest user to go on a date with...